scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/ExtensionMethods.scala

```diff
@@ -0,0 +1,32 @@
+package pl.touk.nussknacker.engine.extension
+
+import java.lang.reflect.Method
+
+object ExtensionMethods {
+
+  private val declarationsWithImplementations = Map[Class[_], Constructor](
+    classOf[Cast] -> new CastConstructor,
+  )
+
+  val registry: Set[Class[_]] = declarationsWithImplementations.keySet
+
+  def applies(clazz: Class[_]): Boolean = registry.contains(clazz)
```

Comment:
You could replace applies and invoke with `PartialFunction`. Thanks to that we would have a lower number of entry points to this class and the interface would be more understandable.